app/backend/src/clj/teet/asset/asset_db.clj

```diff
@@ -511,3 +541,19 @@
       (throw (ex-info "No version creation time found"
                       {:version-id version-id}))
       (d/as-of db c))))
+
+(defn leaf-component?
+  "Is the component a leaf component? A component is a leaf component if
+  the component type does not allow child components."
+  [db component-id]
+  ;; Does there exist a ?ctype such that
+  ;; - ?ctype is the ctype of component-id and
+  ;; - ?ctype is no other ctype's parent
+  (->> (d/q '[:find (some? ?ctype)
```

Comment:
can you use `some?` as an aggregate? it's not documented in the list of builtin aggregates